packages/animations/lib/src/fade_transition.dart

```diff
@@ -0,0 +1,232 @@
+///           child: ConstrainedBox(
+///             constraints: const BoxConstraints(
+///               maxHeight: 300,
+///               maxWidth: 300,
+///               minHeight: 250,
+///               minWidth: 250,
+///             ),
+///             child: const Material(
+///               child: Center(child: FlutterLogo(size: 250)),
+///             ),
+///           ),
+///         ),
+///       ],
+///     );
+///   }
+/// }
+/// ```
+///
+/// The `context` argument is used to look up the [Navigator] for the
+/// modal. It is only used when the method is called. Its corresponding widget
+/// can be safely removed from the tree before the modal is closed.
+///
+/// The `useRootNavigator` argument is used to determine whether to push the
+/// modal to the [Navigator] furthest from or nearest to the given `context`.
+/// By default, `useRootNavigator` is `true` and the modal route created by
+/// this method is pushed to the root navigator.
+///
+/// If the application has multiple [Navigator] objects, it may be necessary to
+/// call `Navigator.of(context, rootNavigator: true).pop(result)` to close the
+/// modal rather than just `Navigator.pop(context, result)`.
+///
+/// The `barrierDismissible` argument is used to determine whether this route
+/// can be dismissed by tapping the modal barrier. This argument defaults
+/// to true. If `barrierDismissible` is true, a non-null `barrierLabel` must be
+/// provided.
+///
+/// The `barrierLabel` argument is the semantic label used for a dismissible
+/// barrier. This argument defaults to "Dismiss".
+///
+/// Returns a [Future] that resolves to the value (if any) that was passed to
+/// [Navigator.pop] when the modal was closed.
+///
+/// See also:
+///
+/// * [FadeModalRoute], which is the route that is built by this function.
+Future<T> showModalWithFadeTransition<T>({
+  @required BuildContext context,
+  bool barrierDismissible = true,
+  String barrierLabel,
+  bool useRootNavigator = true,
+  Widget child,
+}) {
+  barrierLabel = barrierLabel ??
+      MaterialLocalizations.of(context).modalBarrierDismissLabel;
+  assert(useRootNavigator != null);
+  assert(!barrierDismissible || barrierLabel != null);
+  return Navigator.of(context, rootNavigator: useRootNavigator).push<T>(
+    FadeModalRoute<T>(
+      barrierDismissible: barrierDismissible,
+      barrierLabel: barrierLabel,
+      child: child,
+    ),
+  );
+}
+
+/// A modal route that overlays a widget on the current route with the Material
+/// fade transition.
+///
+/// The fade pattern is used for UI elements that enter or exit from within
+/// the screen bounds. Elements that enter use a quick fade in and scale from
+/// 80% to 100%. Elements that exit simply fade out. The scale animation is
+/// only applied to entering elements to emphasize new content over old.
+///
+/// See also:
+///
+/// * [showModalWithFadeTransition], which displays the modal popup.
+class FadeModalRoute<T> extends PopupRoute<T> {
```

Comment:
Wondering if we should just keep this class private and only make showModalWithFadeTransition the entry point?

Reply:
I think my original thought process here was that I felt that the Material library's `_DialogRoute` would have been nice to use here if it were public and more extensible.

Thinking about it now, I guess it wouldn't really matter since we're taking a similar approach as `_DialogRoute` with the `_FadeModalRoute`. I'll change this back to private.